include/f_core/net/network/c_ipv4.h

```diff
@@ -0,0 +1,66 @@
+#ifndef C_IPV4_H
+#define C_IPV4_H
+
+#include <zephyr/posix/arpa/inet.h>
+
+struct net_if; // Forward declaration
+struct device;
+
+class CIPv4 {
+public:
+    static constexpr const char* CLASS_A_NETMASK = "255.0.0.0";
+
+    /**
+     * Constructor
+     * @param ip IP address to initialize
+     */
+    explicit CIPv4(const char* ip);
+
+    /**
+     * Constructor
+     * @param ip IP address to initialize
+     * @param net_iface Network interface to bind to
+     */
+    explicit CIPv4(const char* ip, net_if* net_iface);
+
+    /**
+     * Constructor
+     * @param ip IP address to initialize
+     * @param dev Device to bind to
+     */
+    explicit CIPv4(const char* ip, const device* dev);
+
+    /**
+     * Getter for IP address
+     * @return IP address
+     */
+    [[nodiscard]] const char* GetIp() const { return ip; }
+
+    /**
+     * Getter for address structure
+     * @return Address structure
+     */
+    [[nodiscard]] const in_addr& GetAddr() const { return addr; }
+
+    /**
+     * Getter for error code from initialization
+     * @return Error code
+     */
+    [[nodiscard]] const int GetErr() const { return err; }
```

Comment:
returning const char pointer and const reference above makes sense and having this be a const member function makes sense but did you mean to be returning a const int. Usually returning a const non pointer or reference type is [not what you want](https://clang.llvm.org/extra/clang-tidy/checks/readability/const-return-type.html).

Reply:
Ah yes very right. Had a copypasta moment